File: heap/leetcode_506/solution.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        score_with_index = []
        score_length = len(score)
        result = [None] * score_length 

        for i in range(score_length):
            score_with_index.append([score[i], i])

        score_with_index.sort(key=lambda x: x[0], reverse=True)

        for i in range(score_length):
            index = score_with_index[i][1]
            rank = i + 1

            if rank == 1:
                result[index] = "Gold Medal"
            elif rank == 2:
                result[index] = "Silver Medal"
            elif rank == 3:
                result[index] = "Bronze Medal"
            else:
                result[index] = str(rank)
        
        return result

File: heap/leetcode_506/solution.js
/**
 * @param {number[]} score
 * @return {string[]}
 */
var findRelativeRanks = function(score) {
    const scoreWithIndex = [];
    const scoreLength = score.length;
    const result = new Array(scoreLength);

    for(let i=0; i<scoreLength; i++) {
        scoreWithIndex.push([score[i], i]);
    }

    scoreWithIndex.sort((a, b) => b[0] - a[0]); 

    for (let i=0; i<scoreLength; i++) {
        const index = scoreWithIndex[i][1];
        const rank = i + 1;

        switch(rank) {
            case 1:
                result[index] = "Gold Medal";
                break;
            case 2:
                result[index] = "Silver Medal";
                break;
            case 3:
                result[index] = "Bronze Medal";
                break;
            default:
                result[index] = String(rank);
                break;
        }
    }

    return result;
};